test/lit/passes/global-effects.wast

```diff
@@ -219,6 +219,53 @@
     )
   )
 
+  ;; WITHOUT:      (func $return-call-throw-and-catch (type $0)
+  ;; WITHOUT-NEXT:  (return_call $throw)
+  ;; WITHOUT-NEXT: )
+  ;; INCLUDE:      (func $return-call-throw-and-catch (type $0)
+  ;; INCLUDE-NEXT:  (return_call $throw)
+  ;; INCLUDE-NEXT: )
+  (func $return-call-throw-and-catch
+    (try
+      (do
+        ;; This call cannot be optimized out, as the target throws. However, the
+        ;; surrounding try-catch can be removed even without global effects
+        ;; because the throw from the return_call is never observed by this
+        ;; try-catch.
+        (return_call $throw)
+      )
+      (catch_all)
+    )
+  )
+
+  ;; WITHOUT:      (func $call-return-call-throw-and-catch (type $0)
+  ;; WITHOUT-NEXT:  (try $try
+  ;; WITHOUT-NEXT:   (do
+  ;; WITHOUT-NEXT:    (call $return-call-throw-and-catch)
+  ;; WITHOUT-NEXT:   )
+  ;; WITHOUT-NEXT:   (catch_all
+  ;; WITHOUT-NEXT:    (nop)
+  ;; WITHOUT-NEXT:   )
+  ;; WITHOUT-NEXT:  )
+  ;; WITHOUT-NEXT:  (call $return-call-throw-and-catch)
+  ;; WITHOUT-NEXT: )
+  ;; INCLUDE:      (func $call-return-call-throw-and-catch (type $0)
+  ;; INCLUDE-NEXT:  (call $return-call-throw-and-catch)
+  ;; INCLUDE-NEXT: )
+  (func $call-return-call-throw-and-catch
+    (try
+      (do
+        ;; Even though the body of the previous function is a try-catch_all, the
+        ;; function still throws because of its return_call, so this cannot be
+        ;; optimized out, but once again the entire try-catch can be.
+        (call $return-call-throw-and-catch)
+      )
+      (catch_all)
+    )
+    ;; This cannot be optimized out at all.
+    (call $return-call-throw-and-catch)
```

Comment:
Please test `return_call_indirect/ref` as well, so we have coverage for them adding the `hasReturnCallThrow` field. (IIANM the tests below test the other property, that they do not have `throws_` set and so they can be optimized better.)

Reply:
Added tests, but I'm not sure they test what you're looking for because the callees also have the `calls` effect, which is enough to inhibit optimizations on its own.

Reply:
Yeah, I guess `calls` masks this. Still, those tests look useful - maybe some day we'll make `calls` more refined somehow, and these would become more important.